src/components/mobile-nav-item/mobile-nav-item.tsx

```diff
@@ -0,0 +1,82 @@
+import { useState } from 'react'
+import type { FC } from 'react'
+
+import { useId } from '../../storybook/random-id'
+
+import { Icon } from '../icon'
+
+import { MobileNavItemExpandableProps, MobileNavItemSimpleProps, MobileNavSubItemProps } from './types'
+import {
+  ElMobileNavItemAnchor,
+  ElMobileNavItemBadge,
+  ElMobileNavItemContent,
+  ElMobileNavItemExpanderButton,
+  ElMobileNavItemListItem,
+  ElMobileNavItemUnorderedList,
+  ElMobileNavSubItemUnorderedList,
+} from './styles'
+import { MobileNavSubItem } from './mobile-nav-sub-item'
+
+export type MobileNavItemProps = MobileNavItemSimpleProps | MobileNavItemExpandableProps
+
+const isExpandableVariant = (props: MobileNavItemProps): props is MobileNavItemExpandableProps => {
+  return 'children' in props
+}
+
+type MobileNavItemFC = FC<MobileNavItemProps> & {
+  SubItem: FC<MobileNavSubItemProps>
+}
+
+const MobileNavItem: MobileNavItemFC = (props) => {
+  const [isExpanded, setIsExpanded] = useState(Boolean(props?.isDefaultExpanded))
```

Comment:
**question:** do we need a "default expanded" prop? Also, if `isActive` is true, would we not want the nav item to be expanded?

Reply:
> do we need a "default expanded" prop?

I think it's worthy to have the "default expanded" behavior, so the UI would automatically sync with the passed state by default (in case, when it's on mounting phase)

>  if isActive is true, would we not want the nav item to be expanded?

we could consider this to only rely on single prop that has the right responsibility, seems just by relying the `isActive` is more than enough instead adding the additional one